### PR TITLE
Enhanced Sidebar

### PR DESCRIPTION
**Pull request by rossbulat**
This Enhanced Sidebar PR implements the following:

 * sidebar now transitions between collapsed and expanded views.
 * icon divider is no longer showing in a collapsed view. Reasoning: was hard to notice and the icon spacing was not consistent.
* a toggle has been introduced along the right side of the side bar for easier expanding, more granularity for desktop users.
* tooltips have been introduced to the sidebar in a collapsed view.
* slightly enlarged icons in collapsed view with optimised spacing.

Sidebar Demo

### CODE REVIEW

**Comment by jacogr** [comment]
This is cool! Will take a peek.

File: packages/apps/src/SideBar/Item.tsx [hunk 1]
```diff
@@ -15,27 +15,61 @@ import accountObservable from '@polkadot/ui-keyring/observable/accounts';
...
+  'data-tip': boolean;
+  'data-for': string;
+  'data-tip-disable'?: boolean;
+}
+
 class Item extends React.PureComponent<Props> {
+
+  componentWillUpdate () {
+    ReactTooltip.rebuild();
+  }
```
Comment (jacogr) [inline packages/apps/src/SideBar/Item.tsx]: (nitpick) extra blank line after. (And the one before the method can be removed)

File: packages/apps/src/SideBar/Item.tsx [hunk 2]
```diff
@@ -15,27 +15,61 @@ import accountObservable from '@polkadot/ui-keyring/observable/accounts';
...
 
     if (!this.isVisible()) {
       return null;
     }
 
+    let tooltip: Tooltip = {
+      'data-tip': true,
+      'data-for': `nav-${name}`
+    };
+    isCollapsed
```
Comment (jacogr) [inline packages/apps/src/SideBar/Item.tsx]: Can just do

```js
const tooltip: Tooltip = {
  'data-for': `nav-${name}`,
  'data-tip': true,
  'data-tip-disable': isCollapsed
};
```

Couple of this here -

- `const`
- no need to create `?`` outside the structure
- alphabetical ordering

File: packages/apps/src/SideBar/Item.tsx [hunk 3]
```diff
@@ -15,27 +15,61 @@ import accountObservable from '@polkadot/ui-keyring/observable/accounts';
...
+          {...tooltip}
         >
-          <Icon name={icon} /><span className='text'>{t(`sidebar.${name}`, i18n)}</span>
+          <Icon name={icon} />
+          <span className='text'>{t(`sidebar.${name}`, i18n)}</span>
+          <ReactTooltip
+           id={`nav-${name}`}
+           place='right'
+           offset={ { right: -4 } }
+           effect='solid'
```
Comment (jacogr) [inline packages/apps/src/SideBar/Item.tsx]: Order these alphabetical

File: packages/apps/src/SideBar/index.tsx [hunk 4]
```diff
@@ -109,6 +112,7 @@ class SideBar extends React.PureComponent<Props, State> {
             key={route.name}
             t={t}
             route={route}
+            isCollapsed={isCollapsed}
```
Comment (jacogr) [inline packages/apps/src/SideBar/index.tsx]: (nitpick, maintainability) Alphabetical ordering

**Comment by rossbulat** [comment]
I believe the conflicts are now resolved.

**Pull request merged**